rmldatatfbbox/rmldatatfbbox/core.py

```diff
@@ -5,77 +5,158 @@
+from ravenml.utils.question import user_input, cli_spinner
+from ravenml.data.helpers import default_filter_and_load
+from ravenml.data.write_dataset import write_dataset, write_metadata
+from ravenml.data.options import pass_create
+from ravenml.data.interfaces import CreateInput, CreateOutput
 
 ### COMMANDS ###
-@click.command(help='TensorFlow Object Detection with bounding boxes.')
+@click.group(help='Dataset creation with bounding box info')
 @click.pass_context
 def tf_bbox(ctx):
     pass
+
+@tf_bbox.command(help='Create a dataset.')
+@pass_create
+@click.pass_context
+def create(ctx: click.Context, create: CreateInput):
+
+    cli_spinner("Importing TensorFlow...", _import_od)
+
+    config = create.config.get("plugin")
+
+    # Optional Information from interface (just contains plugin)
+    # metadata = create.plugin_metadata
+
+    # set up TF verbosity
+    if config.get('verbose'):
+        tf.compat.v1.logging.set_verbosity(tf.compat.v1.logging.INFO)
+    else:
+        tf.compat.v1.logging.set_verbosity(tf.compat.v1.logging.FATAL)
     
-# @tf_bbox.command(help='Create a dataset.')
-# def create(ctx):
-#     click.echo("Dataset gets created here.")
-
-# # stdout redirection found at https://codingdose.info/2018/03/22/supress-print-output-in-python/
-# def _import_od():
-#     """ Imports the necessary libraries for object detection training.
-#     Used to avoid importing them at the top of the file where they get imported
-#     on every ravenML command call, even those not to this plugin.
+    # set base directory for dataset 
+    base_dir = create.dataset_path
     
-#     Also suppresses warning outputs from the TF OD API.
-#     """
-#     # create a text trap and redirect stdout
-#     # to suppress printed warnings from object detection and tf
-#     text_trap = io.StringIO()
-#     sys.stdout = text_trap
-#     sys.stderr = text_trap
+    # Transformation functionality is not currently supported, thus the empty array
+    transform_metadata = []
+
+    # Variable type validation is currently not supported
+    dataset_name = config.get("dataset_name") if config.get("dataset_name") else user_input(message="What would you like to name this dataset?")
     
-#     # Calls to _dynamic_import below map to the following standard imports:
-#     #
-#     # import tensorflow as tf
-#     # from object_detection import model_hparams
-#     # from object_detection import model_lib
-#     _dynamic_import('tensorflow', 'tf')
-#     _dynamic_import('object_detection.model_hparams', 'model_hparams')
-#     _dynamic_import('object_detection.model_lib', 'model_lib')
-#     _dynamic_import('object_detection.exporter', 'exporter')
-#     _dynamic_import('object_detection.protos', 'pipeline_pb2', asfunction=True)
+    k_folds_specified = config.get("kfolds") if config.get("kfolds") else user_input(message="How many folds would you like the dataset to have?",default="5")
```

Comment:
I think having ability to do k fold is good, at least if we already have the code to do it. but yes it should be optional and unprompted, similar to the `ec2_policy` field used in the training plugin configs

Reply:
if we are onboard with optional kfold as described here (https://github.com/autognc/ravenML/pull/44#discussion_r456215693) then let's change this to unprompted behavior only controllable via config file. You can use the presence of a "kfold" field to determine if its enabled, and the value of the field can be the # of folds.

Also, maybe good to add a check somewhere to ensure folds < # of images. Otherwise dataset creation will fail on the split logic.